code/modules/antagonists/heretic/status_effects/void_chill.dm

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Contains the "Void Chill" status effect. Harmful debuff which freezes and slows down non-heretics
+ * Cannot affect silicons (How are you gonna freeze a robot?)
+ */
+/datum/status_effect/void_chill
+	id = "void_chill"
+	duration = 30 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/void_chill
+	status_type = STATUS_EFFECT_REFRESH //Custom code
+	on_remove_on_mob_delete = TRUE
+	remove_on_fullheal = TRUE
+	///Current amount of stacks we have
+	var/stacks
+	///Maximum of stacks that we could possibly get
+	var/stack_limit = MAX_FREEZE_STACKS
+	///icon for the overlay
+	var/image/stacks_overlay
+
+/datum/status_effect/void_chill/on_creation(mob/living/new_owner, new_stacks, ...)
+	. = ..()
+	RegisterSignal(owner, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(update_stacks_overlay))
+	set_stacks(new_stacks)
+	owner.add_atom_colour(COLOR_BLUE_LIGHT, TEMPORARY_COLOUR_PRIORITY)
+	owner.update_icon(UPDATE_OVERLAYS)
+
+/datum/status_effect/void_chill/on_apply()
+	if(issilicon(owner))
+		return FALSE
+	return TRUE
+
+/datum/status_effect/void_chill/Destroy()
+	UnregisterSignal(owner, COMSIG_ATOM_UPDATE_OVERLAYS)
+	return ..()
+
+/datum/status_effect/void_chill/on_remove()
+	owner.update_icon(UPDATE_OVERLAYS)
+	owner.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, COLOR_BLUE_LIGHT)
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/void_chill)
+	owner.remove_alt_appearance("heretic_status")
+
+/datum/status_effect/void_chill/tick(seconds_between_ticks)
+	owner.adjust_bodytemperature(-12 * stacks)
+
+/datum/status_effect/void_chill/refresh(mob/living/new_owner, new_stacks, forced = FALSE)
+	. = ..()
+	if(forced)
+		set_stacks(new_stacks)
+	else
+		adjust_stacks(new_stacks)
+	owner.update_icon(UPDATE_OVERLAYS)
+
+///Updates the overlay that gets applied on our victim
+/datum/status_effect/void_chill/proc/update_stacks_overlay(atom/parent_atom, list/overlays)
+	SIGNAL_HANDLER
+
+	if(stacks >= 5)
+		linked_alert.icon_state = "void_chill_oh_fuck"
+		linked_alert.desc = "You had your chance to run, now it's too late. You may never feel warmth again..."
+		linked_alert.update_appearance(UPDATE_ICON_STATE)
```

Comment:
This desc and icon state should be in `/atom/movable/screen/alert/status_effect/void_chill/update_icon_state` and `/atom/movable/screen/alert/status_effect/update_desc`

Then updated via `linked_alert.update_appearance(UPDATE_ICON_STATE|UPDATE_DESC)`